client/client.go

```diff
@@ -35,6 +39,32 @@ var (
 	errorAddingServerCertificateToPool = errors.New("Error adding trusted server certificate to pool.")
 )
 
+var (
+	// 	                  http.Client.Timeout
+	// +--------------------------------------------------------+
+	// +--------+  +---------+  +-------+  +--------+  +--------+
+	// |  Dial  |  |   TLS   |  |Request|  |Response|  |Response|
+	// |        |  |handshake|  |       |  |headers |  |body    |
+	// +--------+  +---------+  +-------+  +--------+  +--------+
+	// +--------+  +---------+             +--------+
+	//  Dial        TLS                     Response
+	//  timeout     handshake               header
+	//  timeout                             timeout
+	//
+	//  It covers the entire exchange, from Dial (if a connection is not reused)
+	// to reading the body. This is to timeout long lasing connections.
+	defaultClientReadingTimeout = 12 * time.Hour
+	// More granular connection timeouts
+	defaultDialTimeout           = 30 * time.Second
+	defaultTLSHandshakeTimeout   = 30 * time.Second
+	defaultResponseHeaderTimeout = 60 * time.Second
+
+	// connection keepalive options
+	connectionKeepaliveTime     = 5 * time.Second
+	connectionKeepaliveInterval = 5 * time.Second
+	connectionKeepaliveProbes   = 10
```

Comment:
See this https://github.com/mendersoftware/mender/pull/181/commits/50936084882b2812bf189fc18fde02714ad19c44#diff-1cd41a2611da49ce63630d5a51bd82c5R41

I should probably add better description there. 
`defaultClientReadingTimeout` is the timeout for the whole connection, from start to after body is read
`defaultDialTimeout` is for establishing the connection 
`defaultTLSHandshakeTimeout` this is quite self explanatory
`defaultResponseHeaderTimeout` is for reading response headers

In some sense those might look like redundant, but those are bit different. As keepalive will eliminate stale connections it will not eliminate extremely slow connections. Those will handle the issue, when connection will be very slow, but active.